src/Compilers/Core/Portable/Operations/Expressions.cs

```diff
@@ -1,405 +1,116 @@
-            visitor.VisitConditionalChoiceExpression(this);
-        }
-
-        public TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            return visitor.VisitConditionalChoiceExpression(this, argument);
-        }
-    }
-
-    internal sealed class Assignment : IExpressionStatement
-    {
-        private readonly AssignmentExpression _assignment;
-
-        public Assignment(IOperation target, IOperation value, SyntaxNode syntax)
+        private static bool IsInvalidOperation(IOperation condition, IOperation ifTrue, IOperation ifFalse, ITypeSymbol resultType)
         {
-            _assignment = new AssignmentExpression(target, value, syntax);
-            this.Syntax = syntax;
-        }
-
-        public SyntaxNode Syntax { get; }
-
-        public OperationKind Kind => OperationKind.ExpressionStatement;
-
-        public bool IsInvalid => _assignment.IsInvalid;
-
-        public IOperation Expression => _assignment;
-
-        public ITypeSymbol Type => null;
-
-        public Optional<object> ConstantValue => default(Optional<object>);
-
-        public void Accept(OperationVisitor visitor)
-        {
-            visitor.VisitExpressionStatement(this);
-        }
-
-        public TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            return visitor.VisitExpressionStatement(this, argument);
-        }
-
-        private sealed class AssignmentExpression : IAssignmentExpression
-        {
-            public AssignmentExpression(IOperation target, IOperation value, SyntaxNode syntax)
-            {
-                this.Value = value;
-                this.Target = target;
-                this.Syntax = syntax;
-            }
-
-            public IOperation Target { get; }
-
-            public IOperation Value { get; }
-
-            public SyntaxNode Syntax { get; }
-
-            public ITypeSymbol Type => this.Target.Type;
-
-            public OperationKind Kind => OperationKind.AssignmentExpression;
-
-            public bool IsInvalid => Target == null || Target.IsInvalid || Value == null || Value.IsInvalid;
-
-            public Optional<object> ConstantValue => default(Optional<object>);
-
-            public void Accept(OperationVisitor visitor)
-            {
-                visitor.VisitAssignmentExpression(this);
-            }
-
-            public TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-            {
-                return visitor.VisitAssignmentExpression(this, argument);
-            }
+            return (condition == null || condition.IsInvalid || ifTrue == null || ifTrue.IsInvalid || ifFalse == null || ifFalse.IsInvalid || resultType == null);
         }
     }
 
-    internal sealed class CompoundAssignment : IExpressionStatement
+    internal sealed partial class ExpressionStatement : IExpressionStatement
```

Comment:
Statements.cs/vb Expressions.cs/vb in Operations directory no longer needed. I just did this to make things build and make this PR to only contains the new Operation Types. those 4 file will be removed in my next PR.